### PR TITLE
Civilian and pirate ship name improvements

### PR DESCRIPTION
**Pull request by Pointedstick**
Added many more words, re-worked a lot of the civilian name generators to reduce awkward combinations, and added comments with example names that should make it easier to see what any given civilian name generator will do.

### CODE REVIEW

File: data/names.txt [hunk 1]
```diff
@@ -691,11 +792,13 @@ phrase "civilian"
 	phrase
 		"male names"
 		"female names"
-		"space nouns"
 		"singular nouns"
```
Comment (jafdy) [inline data/names.txt]: An NPC could be called "Endless Sky"...

**Comment by KiLEdEnNis** [comment]
nice job

File: data/names.txt (hunk 1 above)
Comment (Pointedstick) [inline data/names.txt]: 😉

**Comment by Pointedstick** [comment]
WTF Git

**Comment by Pointedstick** [comment]
Take two: https://github.com/endless-sky/endless-sky/pull/1974

**Pull request closed without merging**